astroid/nodes/node_ng.py

```diff
@@ -248,15 +262,34 @@ def parent_of(self, node):
                 return True
         return False
 
-    def statement(self):
+    @overload
+    def statement(
+        self, future: Literal[False] = ...
+    ) -> Union["nodes.Statement", "nodes.Module"]:
+        ...
+
+    @overload
+    def statement(self, future: Literal[True]) -> Union[NoReturn, "nodes.Statement"]:
+        ...
```

Comment:
`NoReturn` should not be a return type here. As I understand it, it should only be used to annotate functions that **never** return. I.e. unconditional `raise` or `sys.exit`.

Reply:
I added it because I thought it would create problems with `nodes.Module.statement()` if its return type did not match either of the possibilities of `NodeNG`.
`Mypy` is turned off for `astroid`, but won't it complain when a class that inherits from `NodeNG` overwrites a method and returns something that is not included in the possibilities for `NodeNG`?

Reply:
I see what you mean. Tested it with both `mypy` and `pylance` and both seem to except it. Might be that `NoReturn` is special cased somehow.

Reply:
We learn something new every day :)

Reply:
Turns out, you were right all along. Just tested it with strict mode and now `pylance` complains. Should be fixed by adding `NoReturn` just to the function signature in `NodeNG` (not as part of overloads).

Reply:
Added with
https://github.com/PyCQA/astroid/pull/1217/commits/6026560846c00ade3c6f2d2f89f71331300e10e4